photon-server/src/main/java/org/photonvision/common/dataflow/networktables/NTDataPublisher.java

```diff
@@ -0,0 +1,131 @@
+
+        targetPitchEntry = subTable.getEntry("targetPitch");
+        targetAreaEntry = subTable.getEntry("targetArea");
+        targetYawEntry = subTable.getEntry("targetYaw");
+        targetPoseEntry = subTable.getEntry("targetPose");
+//        targetFittedHeightEntry = subTable.getEntry("targetFittedHeight");
+//        targetFittedWidthEntry = subTable.getEntry("targetFittedWidth");
+//        targetBoundingHeightEntry = subTable.getEntry("targetBoundingHeight");
+//        targetBoundingWidthEntry = subTable.getEntry("targetBoundingWidth");
+//        targetRotationEntry = subTable.getEntry("targetRotation");
+    }
+
+    public void setCameraName(String camName) {
+        subTable = rootTable.getSubTable(camName);
+        rawDataEntry = subTable.getEntry("rawData");
+
+        pipelineIndexEntry = subTable.getEntry("pipelineIndex");
+        driverModeEntry = subTable.getEntry("driverMode");
+        latencyMillisEntry = subTable.getEntry("latencyMillis");
+        hasTargetEntry = subTable.getEntry("hasTarget");
+
+        targetPitchEntry = subTable.getEntry("targetPitch");
+        targetAreaEntry = subTable.getEntry("targetArea");
+        targetYawEntry = subTable.getEntry("targetYaw");
+        targetPoseEntry = subTable.getEntry("targetPose");
+//        targetFittedHeightEntry = subTable.getEntry("targetFittedHeight");
+//        targetFittedWidthEntry = subTable.getEntry("targetFittedWidth");
+//        targetBoundingHeightEntry = subTable.getEntry("targetBoundingHeight");
+//        targetBoundingWidthEntry = subTable.getEntry("targetBoundingWidth");
+        //targetRotationEntry = subTable.getEntry("targetRotation")
+    }
+
+    @Override
+    public void accept(CVPipelineResult result) {
+        // send raw data immediately
+        var simplified = new SimplePipelineResult(result);
+        var bytes = simplified.toByteArray();
+        rawDataEntry.forceSetRaw(bytes);
+        rootTable.getInstance().flush();
+
+        // send keyed data after raw
+        // TODO: get pipelineIndex, driverMode
+
+        pipelineIndexEntry.forceSetNumber(pipelineIndexSupplier.get());
+        driverModeEntry.forceSetBoolean(driverModeSupplier.getAsBoolean());
+        latencyMillisEntry.forceSetDouble(result.getLatencyMillis());
+        hasTargetEntry.forceSetBoolean(result.hasTargets());
+
+        // TODO: send all zeroes if no targets?
+        var bestTarget = result.targets.get(0);
+
+        targetPitchEntry.forceSetDouble(bestTarget.getPitch());
+        targetYawEntry.forceSetDouble(bestTarget.getYaw());
+        targetAreaEntry.forceSetDouble(bestTarget.getArea());
+
+        var poseX = bestTarget.getRobotRelativePose().getTranslation().getX();
+        var poseY = bestTarget.getRobotRelativePose().getTranslation().getY();
+        var poseRot = bestTarget.getRobotRelativePose().getRotation().getDegrees();
+        targetPoseEntry.forceSetDoubleArray(new double[] { poseX, poseY, poseRot });
+    }
```

Comment:
I would flush() only at the end -- all the bytepacked stuff will be sent together anyways, and we're rate limited to 100hz

Reply:
Good point, I will change that.

Reply:
Resolved in 71f99d5